### ISSUE TITLE
NRE-C support in the HSSM ecosystem

### ISSUE THREAD

**Issue by cpaniaguam** [lnccbrown/HSSM#1297]
Productionize the NRE-C ratio-estimation work (`ratio_estimation.ipynb` + `nre_lib/` in HSSMSpine), reusing the existing `ssm-simulators` `generate` CLI for data generation.

```mermaid
flowchart LR
    GEN["ssm-simulators\ngenerate CLI"] -->|artifacts| LOAD["artifact loader\n(PR 1)"]
    LOAD --> TRAIN["LANfactory\ntrain RatioApproximator"]
    TRAIN -->|".keras"| PUB["publish to HF hub\n(PR 5)"]
    TRAIN -->|".keras"| INF["inference\nNUTS via NeuralDistribution"]
    PUB -.->|"pretrained, by name"| INF
    INF -->|".nc"| OUT["analysis / tutorial"]
```

## PR 1 — bayesflow-compatible data artifacts

Define the abstraction everything else builds against.

- Artifact schema for NRE training data: parameter draws + one `(rt, choice)` pair each; metadata (model, bounds, seed, ssms version). Prefer npz/parquet over pickle.
- Loader: artifacts → bayesflow-ready dataset (dict / `OfflineDataset`), validated against the ssms model config.
- Note: ssms 0.13.2 has a `ratio_estimator` config stub but no implementation and no bayesflow tooling — this PR writes the missing contract.

**Done when:** schema documented; loader + round-trip test with a tiny committed fixture.

## PR 2 — data generation (ssm-simulators)

- Implement the `ratio_estimator` approach in the `generate` CLI: emit schema-conformant `(theta, rt, choice)` artifacts, skipping the KDE-label step (not needed for NRE).

**Done when:** `generate --config ...` produces artifacts PR 1's loader accepts.

## PR 3 — training + inference (probably LANfactory)

Move the `nre_lib` cores out of HSSMSpine:

- Train a `RatioApproximator` from artifacts (offline; keep online simulation as fallback); save `.keras`.
- Sample posteriors via `NeuralDistribution` + NUTS; save `.nc`. Uniform-over-bounds default priors with per-parameter overrides.
- Preserve `nre_lib`'s fixes: per-call simulator seeding, `workers=1` default, `initvals` support.
- bayesflow/keras stay optional deps.

**Done when:** train + sample work for `ddm` and `angle` with tests; no per-model hardcoding.

## PR 4 — tutorial

- Rewrite `ratio_estimation.ipynb` as a docs tutorial on the new APIs; retire `nre_lib`.
- Reconcile with the existing NRE/LRE tutorials in HSSM (`sbi_nre_integration.ipynb`, `bayesflow_lre_integration.ipynb`) — one coherent story, no overlap.

**Open question — where does this tutorial live?** Depends on how much NRE-specific code HSSM keeps after PR 3.

- **HSSM** — matches the precedent of `sbi_nre_integration.ipynb` / `bayesflow_lre_integration.ipynb`, and fits if HSSM owns the artifact loader (PR 1 consumer side) and the `NeuralDistribution` glue that turns a `.keras` file into a likelihood. Training is shown but imported from LANfactory.
- **LANfactory** — fits if PR 3 lands the whole train+infer stack there and HSSM only re-exports. HSSM then gets a short "using pretrained NREs" page that links out.

Needs sign-off from the ecosystem lead before PR 3 freezes the HSSM/LANfactory boundary.

**Done when:** runs end to end; notebook CI strategy decided (tiny budgets + committed artifacts, or skip list).

## PR 5 — publish trained NREs to Hugging Face

Adapt the publish workflow from `LAN_pipeline_minimal/publish/publish_network.py` (resolve run → stage artifacts → validate → upload → record) for NRE `.keras` artifacts:

- Validation gate before upload (e.g. posterior-recovery smoke check); a skipped gate is not a passed gate.
- Upload layout consumable by HSSM the same way pretrained LANs are (hub repo + canonical filename); record provenance (source run, commit, gate scores).
- Token stays laptop-side; never runs on the cluster.

**Done when:** a trained ratio approximator can be published to the hub and loaded from HSSM by name, with the gate and provenance recorded.

**Comment by cpaniaguam** [comment]
@AlexanderFengler thoughts?

**Comment by AlexanderFengler** [comment]
@cpaniaguam 

few thoughts: 

1. The ssm-simulators side is ok. We want to optimize for some very simple interface / adapter so that generating BayesFlow (and sbi library in the best case) appropriate data comes out. No question about code location here.
2. Training, I think it's ok to first push for a LANfactory update until this works easily, but eventually the bigger question is for me whether we can broaden our notion of "pipeline" so that this training step could also be called from outside LANfactory (think of something like orchestration from Spine / LAN_pipeline_minimal --> depending on what type of network you choose you end up going through LANfactory or not. The HF publish step is kind of affected by this logic too. Should maybe sit outside of LANfactory eventually.
3. On the "Inference" part, we should make sure everything ends HSSM native and we: Generalize HSSM where necessary (I am thinking allowing an "NRE" likelihood kind basically). This section sounds too much like it's targeting a PyMC model directly, but not pushing for enough native HSSM integration.
4. Publishing to HF etc is fine ofc.

---

One other broader thought: Somewhere inbetween the beginning and end (actual prod NREs), we need to pack a bunch of numerical experiments that either convince us to change defaults or allow us to leave some suggestions to users when we think NREs can be appropriate and when not vis a vis LANs.

**Comment by cpaniaguam** [comment]
@DatNoob3
